File: C++/bee 1013.cpp
#include <stdio.h>
#include <iostream>
# include <cmath>
using namespace std;

int maiorABC(int a, int b, int c){
    int maiorAB = (a + b + abs(a-b))/2;
    int maiorABC = (maiorAB + c + abs(maiorAB - c))/2;
    return maiorABC;
}

int main(){
    int a;
    int b;
    int c;
    cin >> a >> b >> c;
    cout << maiorABC(a,b,c) << " eh o maior" << endl;

    return 0;
}

File: Rust/bee_1013.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();

    let valores: Vec<i32> = input
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let maior_ab = (valores[0] + valores[1] + (valores[0] - valores[1]).abs()) / 2;
    let maior_abc = (maior_ab + valores[2] + (maior_ab - valores[2]).abs()) / 2;

    println!("{} eh o maior", maior_abc);
}
